.github/workflows/publish-latest.yml

```diff
@@ -22,10 +22,45 @@ jobs:
         id: has-changesets
         uses: ./.github/actions/has-changesets
 
+  build-test:
+    runs-on: ubuntu-latest
+    needs: setup
+    if: ${{ needs.setup.outputs.has-changesets != 'true' }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      - name: Add Amplify CLI
+        run: yarn global add @aws-amplify/cli
+      - name: Get CLI versions
+        id: cli-version
+        run: echo "::set-output name=version::$(amplify --version)"
+      - name: Create or restore environments cache
+        id: environments-cache
+        uses: actions/cache@v2
+        with:
+          path: canary/environments/**/aws-exports.js
+          key: ${{ runner.os }}-canary-environments-${{ steps.cli-version.outputs.version }}-${{ hashFiles('canary/environments/**/amplify/**') }}
+      - name: Pull down AWS environments
+        if: steps.environments-cache.outputs.cache-hit != 'true'
+        run: yarn pull
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
```

Comment:
Don't you need `environment: ci` for this action to access the `AWS_SECRET_ACCESS_KEY` or does it pick it up from earlier steps?

Reply:
Oh man, thanks for the catch. You're right, my POC run didn't catch this because environments were cached 🤷

Reply:
Done here [b7528a5](https://github.com/aws-amplify/amplify-ui/pull/1882/commits/b7528a5d933bf8c720f204c020b9aed0dbfa736b)